genie-common-internal/src/main/java/com/netflix/genie/common/internal/jobs/JobConstants.java

```diff
@@ -422,6 +432,11 @@
     public static final String JOB_SUBMISSION_DISABLED_DEFAULT_MESSAGE
         = "Job submission is currently disabled. Please try again later.";
 
+    /**
+     * The name of the setup file (for entities that have one) when downloaded in the job directory.
+     */
+    public static final String GENIE_ENTITY_SETUP_SCRIPT_FILE_NAME = "genie_setup.sh";
```

Comment:
One thing we want to be careful of is files appended with `.sh` cause the browser to force a download of said file which people have complained about in the past. They'd prefer it was returned as `text/plain` so that it opens within the browser window. If we're going to start putting the proper extension on these files we'll have to modify the file server code on the server to make sure `.sh` is remapped to `text/plain`

Reply:
This is a browser behavior. Shouldn't we tell those users Genie has no control over browser behavior, but they do?

Moreover, take a look at our current configurations. All setup script i looked at are already named `.sh`. So in practice there is no change of behavior here.

Reply:
Technically it can control the behavior by setting the `Content-Type`.

This is more pertinent for the `run` file as previously it was named `run.sh` till there were many complaints about the behavior and we changed it to `run` and allowed it to be `text/plain`

Reply:
Given how much time we already spend dealing with content-type negotiation corner cases, I'm not a big fan of adding more complexity.
But if we do really want to do it, I'd do it as interceptor that modifies headers when appropriate, rather than as hack in here by giving weird names to thing.